peer-pod-controller/controllers/peerpodconfig_controller.go

```diff
@@ -0,0 +1,402 @@
+func (r *PeerPodConfigReconciler) peerpodCloudProviderIsValid() error {
+	peerpodscm := corev1.ConfigMap{}
+	err := r.Client.Get(context.TODO(), types.NamespacedName{Name: "peer-pods-cm", Namespace: os.Getenv("PEERPODS_NAMESPACE")}, &peerpodscm)
+	if err != nil && k8serrors.IsNotFound(err) {
+		return err
+	} else if err != nil {
+		// Error reading the object - requeue the request.
+		return err
+	} else {
+		ppCloudProviderName = peerpodscm.Data["CLOUD_PROVIDER"]
+		if ppCloudProviderName == "" || !contains(validCloudProviderNames, ppCloudProviderName) {
+			r.Log.Info("CLOUD_PROVIDER=", "ppCloudProviderName", ppCloudProviderName)
+			return errors.New("set CLOUD_PROVIDER name to one supported in cloud-api-provider, found CLOUD_PROVIDER=" + ppCloudProviderName)
+		}
+	}
+	return nil
+}
+
+func MountProgagationRef(mode corev1.MountPropagationMode) *corev1.MountPropagationMode {
+	return &mode
+}
+
+/*
+  cloudProvderName needs to be verified against validCloudProviders by caller
+*/
+func (r *PeerPodConfigReconciler) createCcaDaemonset(cloudProviderName string) *appsv1.DaemonSet {
+	var (
+		runPrivileged           = true
+		runAsUser         int64 = 0
+		defaultMode       int32 = 0600
+		sshSecretOptional       = true
+	)
+
+	dsName := "peer-pod-controller-cca-daemon"
+	dsLabelSelectors := map[string]string{
+		"name": dsName,
+	}
+	nodeSelector := map[string]string{
+		"node-role.kubernetes.io/worker": "",
+	}
+	return &appsv1.DaemonSet{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "apps/v1",
+			Kind:       "DaemonSet",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      dsName,
+			Namespace: os.Getenv("PEERPOD_NAMESPACE"),
+		},
+		Spec: appsv1.DaemonSetSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: dsLabelSelectors,
+			},
+			UpdateStrategy: appsv1.DaemonSetUpdateStrategy{
+				Type: "RollingUpdate",
+				RollingUpdate: &appsv1.RollingUpdateDaemonSet{
+					MaxUnavailable: &intstr.IntOrString{
+						Type:   intstr.Int,
+						IntVal: 1,
+					},
+				},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: dsLabelSelectors,
+				},
+				Spec: corev1.PodSpec{
+					ServiceAccountName: "default",
+					NodeSelector:       nodeSelector,
+					HostNetwork:        true,
+					Containers: []corev1.Container{
+						{
+							Name: "cc-runtime-install-pod",
+							// TODO make configurable via env var
+							Image:           "quay.io/confidential-containers/cloud-api-adaptor-" + cloudProviderName + ":latest",
+							ImagePullPolicy: "Always",
+							SecurityContext: &corev1.SecurityContext{
+								// TODO - do we really need to run as root?
+								Privileged: &runPrivileged,
+								RunAsUser:  &runAsUser,
+							},
+							Command: []string{"/usr/local/bin/entrypoint.sh"},
+							Env: []corev1.EnvVar{
+								{
+									Name:  "PAUSE_IMAGE",
+									Value: "quay.io/bpradipt/okd-pause",
+								},
+							},
+							EnvFrom: []corev1.EnvFromSource{
```

Comment:
Env variables will need to be populated from both secret and configmap.
Example configmap and secret that is used for install - https://github.com/confidential-containers/cloud-api-adaptor/blob/staging/install/overlays/aws/kustomization.yaml

Reply:
added another EnvFromSource for creating environment vars from the configmap